server/src/daemon/service.rs

```diff
@@ -24,11 +24,33 @@ use super::{
     collection::Collection, error::ServiceError, prompt::Prompt, session::Session, Result,
 };
 
+#[derive(Debug)]
+pub struct ServiceManager {
+    sessions: HashMap<OwnedObjectPath, Session>,
+}
+
+impl ServiceManager {
+    pub fn new() -> Self {
+        Self {
+            sessions: HashMap::new(),
+        }
+    }
+
+    pub fn sessions(&self) -> &HashMap<OwnedObjectPath, Session> {
```

Comment:
Don't add a getter that is not used, yet. You would just need an API to get a Option<&Session> from a session path

Reply:
wip.